zk/datastream/server/datastream_populate.go

```diff
@@ -125,20 +125,22 @@ func WriteBlocksToStream(
 			return err
 		}
 
-		gersInBetween, err := reader.GetBatchGlobalExitRootsProto(prevBatchNum, batchNum)
+		nextBatchNum, err := reader.GetBatchNoByL2Block(currentBlockNumber + 1)
 		if err != nil {
 			return err
 		}
 
-		l1InfoMinTimestamps := make(map[uint64]uint64)
+		// a 0 next batch num here would mean we don't know about the next batch so must be at the end of the batch
+		isBatchEnd := nextBatchNum == 0 || nextBatchNum > batchNum
```

Comment:
I wonder if we should have some constant (non 0) for the end of the batch to be returned by GetBatchNoByL2Block. Isn’t batch number 0 a legit batch number?

Reply:
Batch 0 is legit, but it's injected during startup of the node before the chain starts working so it will never be the "next" batch, but signals that we have no data for the batch being queried so we can use it safely to detect the end of the last known batch in the DB

Reply:
I'm not sure what you mean in the first sentence about the constant non 0

Reply:
Either return batch -1 and name it a constant like BATCH_NOT_FOUND, or return 0 and also err = ErrNotFound and check for that.

Reply:
got it, thanks for the clarification

Reply:
I'll use the error approach here as we're dealing with uint and I don't want to use the max value to denote not found

Reply:
In fact, just a bool `found`, it's not an error

Reply:
Sounds good! I think it is also more in Go-style

Reply:
In fact.... :) this creates quite the large ripple, I think a simple `DoesBatchExist` call is the better option

Reply:
That also works well to me.